sdw_updater/UpdaterApp.py

```diff
@@ -222,39 +221,41 @@ def __init__(self):
         QThread.__init__(self)
 
     def run(self):
+        results = {}
 
         # Update dom0 first, then apply dom0 state. If full state run
         # is required, the dom0 state will drop a flag.
         self.progress_signal.emit(5)
-        upgrade_generator = Updater.apply_updates(vms=["dom0"], progress_start=5, progress_end=10)
-
-        results = {}
-        for vm, progress, result in upgrade_generator:
-            results[vm] = result
-            self.progress_signal.emit(progress)
+        results["dom0"] = Updater.apply_updates_dom0()
 
         # apply dom0 state
         self.progress_signal.emit(10)
         # add to results dict, if it fails it will show error message
         results["apply_dom0"] = Updater.apply_dom0_state()
```

Comment:
(If dom0 updates fail, either at the `check` or the `apply` stage, we might bail early, rather than having the user run through the vm updates as well, since they will end up with `UPDATES_FAILED` anyway and since they may be missing key changes from dom0, eg a migration, that would make the template updates stale anyway.)

Reply:
Sorry I didn't see this comment til today - yes, my suggestion was to stop the update routine at the point in the code where I made this comment if the dom0 updates failed, which is similar to your second bullet. I'll take a look at your changes.